src/Core/src/Handlers/Image/ImageHandler.Windows.cs

```diff
@@ -0,0 +1,91 @@
+﻿#nullable enable
+using System.Threading.Tasks;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+
+namespace Microsoft.Maui.Handlers
+{
+	public partial class ImageHandler : ViewHandler<IImage, FrameworkElement>
+	{
+		double _lastScale = 0.0;
+
+		protected Image? RealNativeView { get; set; }
+
+		protected override FrameworkElement CreateNativeView()
+		{
+			RealNativeView = new Image();
+			return new Border { Child = RealNativeView };
+		}
+
+		protected override void ConnectHandler(FrameworkElement nativeView)
+		{
+			base.ConnectHandler(nativeView);
+
+			_lastScale = 0.0;
+			nativeView.Loaded += OnNativeViewLoaded;
+			nativeView.Unloaded += OnNativeViewUnloaded;
+		}
+
+		protected override void DisconnectHandler(FrameworkElement nativeView)
+		{
+			base.DisconnectHandler(nativeView);
+
+			if (nativeView.XamlRoot != null)
+				nativeView.XamlRoot.Changed -= OnXamlRootChanged;
+
+			_lastScale = 0.0;
+			nativeView.Loaded -= OnNativeViewLoaded;
+			nativeView.Unloaded -= OnNativeViewUnloaded;
+
+			_sourceManager.Reset();
+		}
+
+		public static void MapAspect(ImageHandler handler, IImage image) =>
+			handler.RealNativeView?.UpdateAspect(image);
+
+		public static void MapIsAnimationPlaying(ImageHandler handler, IImage image) =>
+			handler.RealNativeView?.UpdateIsAnimationPlaying(image);
+
+		public static void MapSource(ImageHandler handler, IImage image) =>
+			MapSourceAsync(handler, image).FireAndForget(handler);
+
+		public static async Task MapSourceAsync(ImageHandler handler, IImage image)
+		{
+			if (handler.RealNativeView == null)
+				return;
+
+			var token = handler._sourceManager.BeginLoad();
+
+			var provider = handler.GetRequiredService<IImageSourceServiceProvider>();
+			var result = await handler.RealNativeView.UpdateSourceAsync(image, provider, token);
+
+			handler._sourceManager.CompleteLoad(result);
+		}
+
+		void OnNativeViewLoaded(object sender = null!, RoutedEventArgs e = null!)
+		{
+			if (NativeView?.XamlRoot != null)
+			{
+				_lastScale = NativeView.XamlRoot.RasterizationScale;
```

Comment:
Is there a general place this value can get stored? Or maybe an extension method we can make that represents this flow of operations

var token = NAtiveView.WatchForRasterChanges(() => callback);
NativView.StopWatchingForRasterChanges(token)

Reply:
Yes. https://github.com/dotnet/maui/issues/887

This also happens on iOS. I assume Android does not change, but I am probably wrong. I also need to add that. Maybe.